lib/binary_to_decimal.rb

```diff
@@ -4,6 +4,62 @@
 # The least significant bit is at index 7.
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
+
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  decimal_array = []
+  
+  # make sure array is 8 digits. If not, raise an error
+  if binary_array.length != 8
+    raise NotImplementedError.new("Please provide an array with eight elements")
+  end
+  
+  # assign decimal values to binary digits and put them in an array
+  
```

Comment:
This works, but you've hardcoded the powers of 2 here.  You should look for a way to calculate these, maybe using the length of the array.